### ISSUE TITLE
When you upgrade to neo4j-ogm 1.1.4+ QueryResults could break if you use Maps. [DATAGRAPH-822]

### ISSUE THREAD

**Issue by spring-projects-issues** [spring-projects/spring-data-neo4j#1384]
**[Gary Rosales](https://jira.spring.io/secure/ViewProfile.jspa?name=pwfixed)** opened **[DATAGRAPH-822](https://jira.spring.io/browse/DATAGRAPH-822?redirect=false)** and commented

Before neo4j-ogm 1.1.4, we could map the result of a `@Query` to a `@QueryResult` that had a Map<String,String> property. I guess now we need to use the `@Convert` annotation but seeing that it used to work before at least for a simple map of strings to strings, I think it should still support this simple conversion.

Currently using the latest version of neo4j-ogm you get a ClassNotFound exception because it tries to use the parameterized type for the map (java/lang/String;Ljava/lang/String) instead of realizing is a map.

The reason why I didn't file this bug against neo4j-ogm is because this bug is only visible if you are using the GraphRepositories from spring-data. If you just use neo4j-ogm it will give you a warning telling and the query will return an empty list without throwing an exception.

I wrote a simple project that can reproduce the problem:

https://github.com/pointwisefixed/BrokenMapNeo4jOgm

Basically if you use neo4j-ogm 1.1.4+ it breaks, and if you use neo4j-ogm <= 1.1.3, it works.

---

**Affects:** 4.0 GA
